lib/browser/browser.ts

```diff
@@ -395,30 +384,15 @@ class OktaAuthBrowser extends OktaAuthBase implements OktaAuth, SignoutAPI {
     }
   }
 
-  async handleAuthentication(): Promise<void> {
+  /**
+   * Store parsed tokens from redirect url and clean intermediate state
+   */
+  async handleAuthentication(): Promise<string | null> {
     const { tokens } = await this.token.parseFromUrl();
     this.tokenManager.setTokens(tokens);
-  }
-
-  setFromUri(fromUri?: string): void {
-    // Use current location if fromUri was not passed
-    fromUri = fromUri || window.location.href;
-    // If a relative path was passed, convert to absolute URI
-    if (fromUri.charAt(0) === '/') {
-      fromUri = window.location.origin + fromUri;
-    }
-    sessionStorage.setItem(REFERRER_PATH_STORAGE_KEY, fromUri);
-  }
-
-  getFromUri(relative = false): string {
-    let fromUri = sessionStorage.getItem(REFERRER_PATH_STORAGE_KEY) || window.location.origin;
-    sessionStorage.removeItem(REFERRER_PATH_STORAGE_KEY);
-    if (!relative) {
-      return fromUri;
-    }
-
-    const { pathname, search, hash } = getUrlParts(fromUri);
-    fromUri = `${pathname}${search}${hash}`;
+    const storage = browserStorage.getSessionStorage();
+    const fromUri = storage.getItem(REFERRER_PATH_STORAGE_KEY);
+    storage.removeItem(REFERRER_PATH_STORAGE_KEY);
     return fromUri;
```

Comment:
This is a little odd. It looks like downstream SDKs would do something like:

```
handleAuthentication().then(fromUri => window.location.assign(fromUri))
```

Seems like it's not really "handling" everything. The documentation says

```
Parses tokens from the redirect url and stores them.
```

and then describes it resolving to the `fromUri`. Maybe it would be better if this method was named according to what it does (parse tokens and store them). Then the URI handling stuff is done in a separate method also named appropriately. (redirect to saved uri?).

Reply:
I agree we should rename `handleAuthentication` and return void.
For the fromUri handling, I think we can just bring back `getFromUri` and `setFromUri`, then it's up to the downstream SDKs and the apps to use the methods.

What we can do is to add more code samples in readme.